xgi/classes/hypergraph.py

```diff
@@ -910,8 +910,50 @@ def copy(self):
         H : Hypergraph
             A copy of the hypergraph.
 
+        Notes
+        -----
+
+        There is no guarantee that performing similar operations on a hypergraph and its
+        copy after the copy is made will yield the same results.  For example,
+
+        >>> import xgi
+        >>> H = xgi.Hypergraph([[1, 2, 3], [4], [5, 6], [6, 7, 8]])
+        >>> H.add_edge([1, 3, 5], id=10)
+        >>> K = H.copy()
+        >>> H.add_edge([2, 4]); K.add_edge([2, 4]);
+        >>> list(H.edges) == list(K.edges)
+        False
+
+        The difference is the IDs assigned to new edges:
+
+        >>> H.edges
+        EdgeView((0, 1, 2, 3, 10, 4))
+        >>> K.edges
+        EdgeView((0, 1, 2, 3, 10, 11))
+
         """
-        return subhypergraph(self)
+        copy = self.__class__()
+        nn = self.nodes
+        copy.add_nodes_from((n, deepcopy(attr)) for n, attr in nn.items())
+        ee = self.edges
+        copy.add_edges_from(
+            (ee.members(e), e, deepcopy(attr)) for e, attr in ee.items()
```

Comment:
This all looks good, but I'm wondering if might be faster to call members once:
```
copy.add_edges_from((e, id, deepcopy(self.edges[id])) for id, e in ee.members(dtype=dict))
```

Reply:
Good catch, fixed!